src/SprykerEco/Zed/Unzer/Business/Payment/Processor/Refund/MarketplaceRefundProcessor.php

```diff
@@ -0,0 +1,224 @@
+     * @param \SprykerEco\Zed\Unzer\Business\Payment\Mapper\UnzerPaymentMapperInterface $unzerPaymentMapper
+     * @param \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerPaymentAdapterInterface $unzerPaymentAdapter
+     * @param \SprykerEco\Zed\Unzer\Business\Payment\Saver\UnzerPaymentSaverInterface $unzerPaymentSaver
+     */
+    public function __construct(
+        UnzerReaderInterface $unzerReader,
+        UnzerRefundAdapterInterface $unzerRefundAdapter,
+        UnzerPaymentMapperInterface $unzerPaymentMapper,
+        UnzerPaymentAdapterInterface $unzerPaymentAdapter,
+        UnzerPaymentSaverInterface $unzerPaymentSaver
+    ) {
+        $this->unzerReader = $unzerReader;
+        $this->unzerRefundAdapter = $unzerRefundAdapter;
+        $this->unzerPaymentMapper = $unzerPaymentMapper;
+        $this->unzerPaymentAdapter = $unzerPaymentAdapter;
+        $this->unzerPaymentSaver = $unzerPaymentSaver;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function refund(RefundTransfer $refundTransfer, OrderTransfer $orderTransfer, array $salesOrderItemIds): void
+    {
+        $paymentUnzerTransfer = $this->unzerReader->getPaymentUnzerByOrderReference($orderTransfer->getOrderReference());
+        $unzerRefundTransfers = $this->createUnzerMarketplaceRefundTransfers($refundTransfer, $paymentUnzerTransfer);
+
+        foreach ($unzerRefundTransfers as $unzerRefundTransfer) {
+            $this->unzerRefundAdapter->refundPayment($unzerRefundTransfer);
+        }
+
+        $this->saveUnzerPaymentDetails($paymentUnzerTransfer, $salesOrderItemIds);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\RefundTransfer $refundTransfer
+     * @param \Generated\Shared\Transfer\PaymentUnzerTransfer $paymentUnzerTransfer
+     *
+     * @return array<\Generated\Shared\Transfer\UnzerRefundTransfer>
+     */
+    protected function createUnzerMarketplaceRefundTransfers(
+        RefundTransfer $refundTransfer,
+        PaymentUnzerTransfer $paymentUnzerTransfer
+    ): array {
+        $refundTransfer = $this->setParticipantIdForRefundItems($refundTransfer);
+        $participantReorderedItems = $this->reorderRefundItemsByParticipant($refundTransfer);
+
+        $unzerRefundTransfers = [];
+        foreach ($participantReorderedItems as $participantId => $itemTransfers) {
+            $unzerRefundTransfers[] = $this->buildUnzerMarketplaceRefundTransfer(
+                $paymentUnzerTransfer,
+                $participantId,
+                $itemTransfers,
+            );
+        }
+
+        return $unzerRefundTransfers;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaymentUnzerTransfer $paymentUnzerTransfer
+     * @param string $participantId
+     * @param array<\Generated\Shared\Transfer\ItemTransfer> $itemTransfers
+     *
+     * @return \Generated\Shared\Transfer\UnzerRefundTransfer
+     */
+    protected function buildUnzerMarketplaceRefundTransfer(
```

Comment:
`buildUnzerMarketplaceRefundTransfer()`->  `createUnzerRefundTransfer()`

Reply:
Why still `createUnzerMarketplaceRefundTransfer()` if we return `UnzerRefundTransfer`?